bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/WriteEntryProcessorV3.java

```diff
@@ -170,4 +170,18 @@ public void safeRun() {
                          requestProcessor.addRequestStats);
         }
     }
+
+    /**
+     * this toString method filters out body and masterKey from the output.
+     * masterKey contains the password of the ledger and body is customer data,
+     * so it is not appropriate to have these in logs or system output.
+     */
+    @Override
+    public String toString() {
+        Request.Builder reqBuilder = Request.newBuilder().setHeader(request.getHeader());
```

Comment:
rather than constructing another request without masterkey. why not formatting the string itself (which it would void building another object)?

e.g.

```
@Override
public String toString() {

return MoreObjects.toStringHelper()
      .add("ledgerId", request.getLedgerId())
      ...
      .toString();

}
```

Reply:
I've mixed opinion. 

AFAIU These kind of log lines (involving toString of ProcessorV3) are called rarely and may be it is ok to create a new request object. Also, it is easier to backlist one are two unwanted fields rather than whitelisting all other fields (including header/fields of header).

Reply:
1. Response goes through the same OrderedSafeExecutor and can get logged similarly. Luckily, in the cases I see when response is wrapped into Runnable we also redefine toString for runnable.

2. BookieProtocol.Request defines toString as:

```
public String toString() {
            return String.format("Op(%d)[Ledger:%d,Entry:%d]", opCode, ledgerId, entryId);
        }
```

This should be enough, maybe add tx id to that